server/methods/browseChannels.js

```diff
@@ -78,7 +82,7 @@ const getChannels = (user, canViewAnon, searchTerm, sort, pagination) => {
 	const total = result.count(); // count ignores the `skip` and `limit` options
 	const results = result.fetch().map((room) => {
 		if (room.teamId) {
-			const team = teams.find((team) => team._id === room.teamId);
+			const team = teams.find((team) => team._id === room.teamId) || userTeams.find((team) => team._id === room.teamId) || Promise.await(Team.getInfoById(room.teamId));
 			if (team) {
 				room.belongsTo = team.name;
```

Comment:
In here, a Team's name is the same as the Room's name. So you don't need to fetch team's data to know its name.

Reply:
I think it's not possible to use only `room.name` here since all rooms in `result` are **not** team mains. Which means that the `belongsTo` field from each room needs to be filled with its parent room's name 🤔

Reply:
But there should be an indicator if the room is the "main" room (teamMain i guess). So you can extract the names of the main rooms and use them to match all the other rooms

Something like:
1. Filter the main rooms and store them as an array
2. Looping thorugh all rooms, and using arr.find on the prev array to get the team's name (which is the room's name)

Reply:
Great, certainly better than searching for the team's name in three lists 😆